hyperspy/io_plugins/bruker.py

```diff
@@ -938,9 +942,12 @@ def parse_hypermap(self, index=None,
             skimage.measure, the parser populates reduced array by suming
             results of pixels, thus having lower memory requiriments. Default
             is 1.
-        cutoff_at_kV : None or float
-            Value in keV to truncate the array at. Helps reducing size of
-            array. Default is None.
+        cutoff_at_kV : None, float, int or str
+            Value or method to truncate the array at energy in kV. Helps reducing size of
+            array. Default is None (does not truncate). Numerical value is in kV.
+            "auto" truncates to the last non zero channel (should not be used with slices).
+            "lowest_constraint" truncates to instruments acceleration voltage
+            (should be used only if xrays were generated using electron bombardment).
```

Comment:
I hate errors, users hate errors. Errors should be thrown as last option if other does not work. I think in case lowest constraint is used it should check for hv, and if it is ==0, or if `.hv` instance is not here then fallback to channel number. That would prevent uXRF people of shooting self to the foot.

I also remind myself another scenario where `lowest_constraint` would not be good to use: "slicing" depth of the xray excitation with increasing/decreasing HV. I should also add that warning to the docstrings.

Reply:
or actually, `lowest_constraint` is not clever and redundant. I should replace it with simple `em_hv` (electron microscope high voltage) - simple to remember,  short to type, clear usage case. if no hv is present - then fallback to None and throw warnning.

Reply:
> I hate errors, users hate errors. Errors should be thrown as last option if other does not work. I think in case lowest constraint is used it should check for hv, and if it is ==0, or if `.hv` instance is not here then fallback to channel number. That would prevent uXRF people of shooting self to the foot.

Well, raising error has the advantage of being transparent with users and avoid behaviour that users may find surprising - it is always possible to find someone who be surprised. :D

Reply:
Ok, I remade `auto` to `last_non-zero`; and `lowest_constraint` to `em_hv` (electron microscope HV); which I think is more self explanatory. Now I need to adapt rest of files to that.